packages/slice-machine/pages/changes.tsx

```diff
@@ -36,6 +39,11 @@ const changes: React.FunctionComponent = () => {
 
   const [changesPushed, setChangesPushed] = useState<string[]>([]);
   const [error, setError] = useState<SyncError | null>(null);
+  useEffect(() => {
+    return () => {
+      closeScreenshotsModal();
```

Comment:
This is a different problem that I tried to explain previously: because we split state from local page, if the user switches pages while the modal is open, the state of the modal does stay open, although the component is unmounted. When they come back to the page, the modal is mounted and opened. This callback is called on unmount and prevents this to happen. Try it for yourself!

Reply:
The user shouldn't be able to switch state without clicking outside of the modal.

Reply:
cmd + arrow_left

Reply:
Agreed, you can’t consider only mouse actions.